### ISSUE TITLE
Add more supported LaTeX equation formats

### ISSUE THREAD

**Issue by N0rbert** [marktext/marktext#1188]
### Describe your feature request

Many scientific texts contain equations. Currently texts are sometimes written in RMarkdown.

Some texts are books and articles written in [bookdown](https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#equations), some of texts are presentations written with [xaringan](https://bookdown.org/yihui/rmarkdown/xaringan.html). 

### What problem does this feature solve?

Support many LaTeX equation notations such as:

```
Block LaTeX equation without auto-numbering:

$$ g(x) = \cos(x)$$

---

Environment `\begin{equation}...\end{equation}`
with auto-numbering:

\begin{equation}
h(z) = \sinh(z)
\end{equation}

---

Environment `\begin{equation}...(\#eq:label)\end{equation}`
with auto-numbering and label (for bookdown):

\begin{equation}
  f\left(k\right) = \binom{n}{k} p^k\left(1-p\right)^{n-k}
  (\#eq:binom)
\end{equation}

---

Environment `\begin{equation}...\label{label}\end{equation}`
with auto-numbering label for xaringan

\begin{equation}
  g\left(k\right) = \binom{n}{k} p^k\left(1-p\right)^{n-k}
\label{binom2}
\end{equation}
```

### Additional context

In this feature request I ask only for rendering of equations. 
Auto-numbering is not needed for me.
I'll do it in RStudio after knitting of the document.

**Comment by fxha** [comment]
@N0rbert Do you mean that we should allow LaTeX outside math blocks and also in inline code? Are there any markdown standardizations? Or that we should improve the overall LaTeX function support?

**Comment by N0rbert** [comment]
@fxha thanks for reply. The inline math is already supported, but math blocks (as shown above) are not supported. You can install [ReText](https://github.com/retext-project/retext) to get the taste of what I mean. This project [uses MathJax](https://github.com/retext-project/retext/wiki/Math) extension to show equations.
I have tested many markdown editors and I see that yours is the best alternative for proprietary Typora. So your users will be very thankful if you add such functionality.

**Comment by fxha** [comment]
reStructuredText and Markdown are to different markup languages (#1221). We currenlty using `$ ... $` for inline math and double `$` for display math:

```
$$
...
$$
```

A table of all supported functions can be found [here](https://katex.org/docs/supported.html).

@Jocs I guess we need a end-user documentation with all supported features, notation etc.

**Comment by Jocs** [comment]
>@Jocs I guess we need a end-user documentation with all supported features, notation etc.

Yes, we can add user document in the official website, this will be done maybe in the v0.17.0. I think it maybe include bellow parts.

1. Quick start
2. How to use some markdown syntax in Mark Text
3. Advanced features, custom themes, upload image settings, etc.

**Comment by fbruetting** [comment]
By the way: These blocks are also the format of ConTeXt, which is like a much cleaner successor of LaTeX. So this way we could get some ConTeXt compatibility. 😎️

**Comment by peverati** [comment]
I definitely second this request. I'm searching for a WYSIWYG markdown editor for academic use that can render equations using the

\begin{equation} 
...
\end{equation}

tags that are the only way of getting numbered equations in bookdown. This will put Mark Text ahead of the competition, including Typora!

**Comment by YuuuuXie** [comment]
@fxha @Jocs Not sure how Typora implemented all these things, but definitely it supports a few features which are very very useful for equations:
https://support.typora.io/Math/#tex-commands-available-in-typora

And it is still using $$...$$ without `\begin{equation}...\end{equation}`.

For example, the most important and attractive features for me are 
- new commands using \def or\newcommand
- cross reference
- auto numbering

as I usually need to type lots of equations in one doc

**Comment by N0rbert** [comment]
Still not fixed in 0.16.3.

**Comment by zymawy** [comment]
this feature highly recommended

**Comment by N0rbert** [comment]
Still not fixed in 0.20.0~beta.4.
